src/Execution/Execution.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+namespace Digia\GraphQL\Execution;
+
+use Digia\GraphQL\Error\GraphQLError;
+use Digia\GraphQL\Language\AST\Node\DocumentNode;
+use Digia\GraphQL\Language\AST\Node\FieldNode;
+use Digia\GraphQL\Language\AST\Node\OperationDefinitionNode;
+use Digia\GraphQL\Language\AST\Node\SelectionSetNode;
+use Digia\GraphQL\Language\AST\NodeKindEnum;
+use Digia\GraphQL\Type\Definition\ObjectType;
+use Digia\GraphQL\Type\Schema\Schema;
+
+/**
+ * Class Execution
+ * @package Digia\GraphQL\Execution
+ */
+class Execution
+{
+    /**
+     * @var ExecutionContext
+     */
+    protected $context;
+
+    /**
+     * Execution constructor.
+     * @param ExecutionContext $context
+     */
+    public function __construct(ExecutionContext $context)
+    {
+        $this->context = $context;
+    }
+
+    /**
+     * @param Schema $schema
+     * @param DocumentNode $documentNode
+     * @param null $rootValue
+     * @param null $contextValue
+     * @param null $variableValues
+     * @param null $operationName
+     * @param callable|null $fieldResolver
+     * @return ExecutionResult
+     */
+    public static function execute(
+        Schema $schema,
+        DocumentNode $documentNode,
+        $rootValue = null,
```

Comment:
Can't we type the following 4 args? Or at least mention that they are `mixed` in the docblock?